### ISSUE TITLE
salt-minion does not clean up temp files for templates

### ISSUE THREAD

**Issue by yhekma** [saltstack/salt#37541]
### Description of Issue/Question
When rendering templates the temporary files in /tmp are not cleaned up so eventually you will either run out of disk space or inodes (probably the latter)

### Steps to Reproduce Issue
Create some files on a minion using templates and observe the /tmp directory filling up after each `salt-call state.apply`

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2016.3.4

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-100-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by yhekma** [comment]
@gtmanfred Can you say anything about when or if this will be addressed? This problem is really hurting us because of the frequent salt runs and we have accumulated some 20k files in 24 hours per server. I think that if there are people out there that use any kind of automated testing setup thing are going to look really bad for them pretty soon, especially since these files get created even when you pass `test=True`.

**Comment by gtmanfred** [comment]
Your pull request will go into 2016.3 branch so the change will be in 2016.3.5.  We have a release candidate for carbon set, so it won't go into that, and unless we make a new release candidate which we are hoping we don't have to do, it won't be in 2016.11.0, but will be in 2016.11.1.

**Comment by yhekma** [comment]
@gtmanfred So if I understand correctly there is a good chance that 2016.11 will be released with this bug preset? If that is the case, I must say I find that troubling to say the least. Could you please elaborate on available paths to go about getting this fixed asap both in 2016.3 and upcoming releases?

**Comment by gtmanfred** [comment]
Can you give me an example state so that I can check with 2016.11.0rc2?

Just a regular file.managed with a template?

**Comment by rickh563** [comment]
ZD-1048

**Comment by yhekma** [comment]
@gtmanfred Yeah, just a regular file managed by a template. When doing `salt-call state.apply` you can observe the file in /tmp/ doesn't get cleaned up.

**Comment by aboe76** [comment]
@gtmanfred it gets worse, I'm running salt minion masterless from development version,
```
Salt Version:
               Salt: 2016.11.0-360-g2c6a874
```
with an stack as ext_pillar, and my pillar data is rendered in /tmp/tmp***** files.
and it stays there until I clean it.

**Comment by aboe76** [comment]
I can confirm that https://github.com/saltstack/salt/pull/37540 fixes it in latest development version from git

**Comment by danlsgiga** [comment]
I've just noticed the same behaviour on all my minions... thousands of tmpXXXX files on /tmp. Hope this to be addressed sooner than later.

**Comment by cachedout** [comment]
We've got a fix in now.

**Comment by aboe76** [comment]
@cachedout will this https://github.com/saltstack/salt/pull/37540 be merged in develop too?

**Comment by gtmanfred** [comment]
It will be forward merged when we forward merge to 2016.11 and then develop

**Comment by cachedout** [comment]
I don't believe there's additional work here so I'm going to go ahead and close this. As @gtmanfred has already stated, this fix will make its way forward.

cc: @meggiebot and @rallytime  to the fact that #37540 needs to get into the final release of Carbon, as already discussed.
